client/lib/domains/dns/index.js

```diff
@@ -126,7 +130,49 @@ function getFieldWithDot( field ) {
 	return ( typeof field === 'string' && field.match( /^([a-z0-9-_]+\.)+\.?[a-z]+$/i ) ) ? field + '.' : field;
 }
 
+function isWpcomRecord( record ) {
+	return /^wpcom:/.test( record.id );
```

Comment:
This looks like a job for [`startsWith`](https://lodash.com/docs#startsWith)! :stuck_out_tongue_winking_eye: Both will work, obviously, but `startsWith` shows the intent of this function better than trying to grok a regexp (however simple it may be), IMHO.